src/naming.jl

```diff
@@ -250,3 +250,89 @@ end
 function dict2ntuple(dict::Dict{Symbol, T}) where T
     NamedTuple{Tuple(keys(dict))}(values(dict))
 end
+
+
+"""
+    parse_savename(filename::AbstractString; kwargs...)
+                        parsetypes = (Int, Float64),
+                        connector = "_")
+Try to convert a shorthand name produced with [`savename`](@ref) into a dictionary 
+containing the parameters, a prefix and suffix string.
+
+## Keywords
+* `connector = "_"` : string used to connect the various entries.
+* `parsetypes = (Int, Float64)` : tuple used to define the types which should
+  be tried when parsing the values given in `filename`. Fallback is `String`.
+"""
+function parse_savename(filename::AbstractString;
+                        parsetypes = (Int, Float64),
+                        connector = "_")
+    # Prefix can be anything, so it might also contain a folder which's
+    # name was generated using savename. Therefore first the path is split
+    # into folders and filename.
+    prefix_part, savename_part = dirname(filename),basename(filename)
+    # Extract the suffix. A suffix is identified by searching for the last "."
+    # after the last "=".
+    last_eq = findlast("=",savename_part)
+    last_dot = findlast(".",savename_part)
+    if last_dot == nothing || last_eq > last_dot
+        # if no dot is after the last "="
+        # there is no suffix
+        name, suffix = savename_part,""
+    else
```

Comment:
Wow, this is very impressive; you have implemented a very smart function, congrats! :)